### PR TITLE
ut: Add comprehensive unit tests for PlatformHistoryServiceImpl

### PR DESCRIPTION
**Pull request by msslulu**
English | [简体中文](https://github.com/opentiny/tiny-engine-backend-java/blob/main/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine-backend-java/blob/main/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced input validation for platform history operations to reject invalid IDs and improperly formatted names or versions.

* **Tests**
  * Added comprehensive test suite covering platform history queries, creation, updates, and deletion with extensive edge case and error handling scenarios.

### CODE REVIEW

**Review by hexqi — APPROVED** [review]

File: base/src/main/java/com/tinyengine/it/service/platform/impl/PlatformHistoryServiceImpl.java [hunk 1]
```diff
@@ -51,6 +51,9 @@ public List<PlatformHistory> queryAllPlatformHistory() {
      */
     @Override
     public PlatformHistory queryPlatformHistoryById(Integer id) {
+        if(id == null||id<=0) {
+            throw new IllegalArgumentException("id cannot be null");
+        }
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/main/java/com/tinyengine/it/service/platform/impl/PlatformHistoryServiceImpl.java:56]: no  need fix

File: base/src/test/java/com/tinyengine/it/service/platform/impl/PlatformHistoryServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+		mockPlatformHistory.setName("A".repeat(256)); // Simulate a long name
+		mockPlatformHistory.setVersion("1.0");
+		mockPlatformHistory.setMaterialHistoryId(100);
+
+		Result<PlatformHistory> result = platformHistoryServiceImpl.createPlatformHistory(mockPlatformHistory);
+
+		assertNotNull(result);
+		assertFalse(result.isSuccess());
+		assertEquals(ExceptionEnum.CM008.getResultCode(), result.getCode());
+	}
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/service/platform/impl/PlatformHistoryServiceImplTest.java:597]: no need fix

File: base/src/test/java/com/tinyengine/it/service/platform/impl/PlatformHistoryServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+		mockPlatformHistory.setName("A".repeat(300)); // Simulate a name exceeding max length
+		mockPlatformHistory.setVersion("1.0");
+		mockPlatformHistory.setMaterialHistoryId(100);
+
+		Result<PlatformHistory> result = platformHistoryServiceImpl.createPlatformHistory(mockPlatformHistory);
+
+		assertNotNull(result);
+		assertFalse(result.isSuccess());
+		assertEquals(ExceptionEnum.CM008.getResultCode(), result.getCode());
+	}
```
[the remark this answers is not in the document]
Comment (msslulu) [reply base/src/test/java/com/tinyengine/it/service/platform/impl/PlatformHistoryServiceImplTest.java:666]: no need fix
